internal/featureflags/featureflags.go

```diff
@@ -0,0 +1,75 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package featureflags
+
+import "os"
+
+// FeatureFlag represents a command line switch to turn certain features
+// ON or OFF. Values are read from environment variables defined by
+// the feature flag. The feature flag can have default values that are used
+// when the resp. environment variable does not exist
+// Feature flags are considered to be "enabled" if their resp. environment variable
+// is set (to some value, usually "1").
+// Feature flags are considered to be "disabled" if their resp. environment variable
+// is set to "0" explicitly.
+type FeatureFlag struct {
+	// envName is the environment variable name
+	// that is used to read the value from
+	envName string
+	// defaultEnabled states whether this feature flag
+	// is enabled or disabled by default
+	defaultEnabled bool
+}
+
+// Eval evaluates the feature flag
+func (ff FeatureFlag) Eval() bool {
+	if val, ok := os.LookupEnv(ff.envName); ok {
+		return val != "0" && val != ""
+	}
+	return ff.defaultEnabled
+}
+
+// Env gives back the environment variable name for
+// the feature flag
+func (ff FeatureFlag) Env() string {
+	return ff.envName
+}
+
+// FeatEntities returns the feature flag that tells whether Dynatrace Entities download/matching is enabled or not
+func FeatEntities() FeatureFlag {
+	return FeatureFlag{
+		envName:        "MONACO_FEAT_ENTITIES",
+		defaultEnabled: false,
+	}
+}
+
+// FeatDangerousCommands returns the feature flag that tells whether dangerous commands for the CLI are enabled or not
+func FeatDangerousCommands() FeatureFlag {
+	return FeatureFlag{
+		envName:        "MONACO_ENABLE_DANGEROUS_COMMANDS",
+		defaultEnabled: false,
+	}
+}
+
+// FeatVerifyEnvType returns the feature flag that tells whether the environment check
+// at the beginning of execution is enabled or not
+func FeatVerifyEnvType() FeatureFlag {
```

Comment:
What about calling a 'new' function here that already resolves the value?

Reply:
I don't see a reason for that